### ISSUE TITLE
unless x in y compiles to if (x in y)

### ISSUE THREAD

**Issue by graingert** [decaffeinate/decaffeinate#207]
```
      unless box in @all
        @boxes.push box
        @all.push box
        if @stopped or @disabled()
          @resetStyle()
        else
          @applyStyle(box, true)
        @scrolled = true
```

becomes

```js
    for (let i = 0; i < iterable.length; i++) {
      let box = iterable[i];
      if (__in__(box, this.all)) {
        this.boxes.push(box);
        this.all.push(box);
        if (this.stopped || this.disabled()) {
          this.resetStyle();
        } else {
          this.applyStyle(box, true);
        }
        this.scrolled = true;
      }
    }
```

should be

```js
    for (let i = 0; i < iterable.length; i++) {
      let box = iterable[i];
      if (!__in__(box, this.all)) {
        this.boxes.push(box);
        this.all.push(box);
        if (this.stopped || this.disabled()) {
          this.resetStyle();
        } else {
          this.applyStyle(box, true);
        }
        this.scrolled = true;
      }
    }
```

**Comment by graingert** [comment]
(this is the negation of what it should be)

**Comment by hpurmann** [comment]
That's unfortunate. Good catch. [Minimal REPL example](http://decaffeinate.github.io/decaffeinate/repl/#?evaluate=true&code=unless%20a%20in%20b%0A%20%20b.push%20a)

**Comment by hpurmann** [comment]
Found [another one](http://decaffeinate.github.io/decaffeinate/repl/#?evaluate=true&code=if%20not%20(a%20in%20b)%0A%20%20b.push%20a%0A).

**Comment by eventualbuddha** [comment]
Thanks. This is probably just that `InOp` needs to handle negation properly. PRs welcome! Otherwise I'll get to it Monday.

**Comment by qfox** [comment]
```
unless x instanceof y
  z
```
->
```
if (x instanceof y) {
  z;
}
```

:crying_cat_face:

**Comment by Eden-Sun** [comment]
The problem is conditional parser and InOp parser both handled the negation.

**Comment by eventualbuddha** [comment]
I believe all the examples raised on this issue are now fixed in v2.9.1.
